src/query/frontend/ast/ast.hpp

```diff
@@ -2069,6 +2069,27 @@ class Query : public memgraph::query::Tree, public utils::Visitable<QueryVisitor
   friend class AstStorage;
 };
 
+struct IndexHint {
+  static const utils::TypeInfo kType;
+  const utils::TypeInfo &GetTypeInfo() const { return kType; }
+
+  enum class IndexType { LABEL, LABEL_PROPERTY };
+
+  memgraph::query::IndexHint::IndexType index_type_;
+  memgraph::query::LabelIx label_;
+  std::optional<memgraph::query::PropertyIx> property_ = std::nullopt;
```

Comment:
maybe better for aesthetics

`std::optional<memgraph::query::PropertyIx> property_ = std::nullopt;`